packages/components/forms/src/text-input/TextInput.styles.ts

```diff
@@ -0,0 +1,31 @@
+import { css } from 'emotion';
+import tokens from '@contentful/f36-tokens';
+
+const getStyles = ({ isInvalid, isDisabled }) => ({
+  inputWithCopyButton: css({
+    borderRadius: `${tokens.borderRadiusMedium} 0 0 ${tokens.borderRadiusMedium}`,
+  }),
+  copyButton: css({
+    button: css({
+      borderLeft: 'none',
+      height: '100%',
+      borderRadius: `0 ${tokens.borderRadiusMedium} ${tokens.borderRadiusMedium} 0`,
+
+      '&:focus': {
+        zIndex: tokens.zIndexDefault,
+        borderColor: isInvalid
+          ? tokens.red600
+          : isDisabled
+          ? tokens.gray300
+          : tokens.blue600,
```

Comment:
My suggestion here would be to replace the ternary with a function like this

```ts
const getBorderStyles = (isDisabled, isInvalid) => {
  if (isDisabled) {
    return tokens.gray300
  }

  if (isInvalid) {
    return tokens.red600
  }

  return tokens.blue600
}

// ...

borderColor: getBorderColor(isDisabled, isInvalid)
```

I think I saw this pattern in other components or something similar, btw
The reason for my suggestion is that it's easier to maintain and it returns early when the input is disabled

Reply:
Actually, I have another suggestion... hahhahaah

This is how I'm doing in `TextArea`, I defined one class for `error` and `disabled` with the correct styles
then in the component I do this:

```jsx
className={cx(styles.textarea, {
  [styles.error]: Boolean(validationMessage),
  [styles.disabled]: Boolean(isDisabled)
})}
```

Reply:
Actually, I have another suggestion... hahhahaah

This is how I'm doing in `TextArea`, I defined one class for `error` and `disabled` with the correct styles
then in the component I do this:

```jsx
className={cx(styles.textarea, {
  [styles.error]: Boolean(validationMessage),
  [styles.disabled]: Boolean(isDisabled)
})}
```

Reply:
yeah, good idea, did it the way you suggested.

Reply:
Also the `css` function accepts an array, so you could do something similar on the `.styles.ts` file, the `GhostCheckbox` has an example of that approach:
https://github.com/contentful/forma-36/blob/e2e1471ac8c59a9b91c155f075ddf57f3d8e34c6/packages/components/forms/src/base-checkbox/GhostCheckbox.styles.ts#L64